lib/mods/replace-get-set.js

```diff
@@ -25,25 +25,60 @@ function isDataProperty(type, propertyName) {
 }
 
 /**
- * @param {Node} node - the current ts-morph AST Node
+ * @param {Node} node - the ts-morph AST Node to evaluate
  *
- * @return {boolean} whether the node is a global get
+ * @return {boolean} whether the node is a global property access
  */
-function isGlobalGet(node) {
+function isGlobalPropertyAccess(node) {
   return nodeTypes.isPropertyAccessExpression(node) &&
     !nodeTypes.isCallExpression(node.getParent()) &&
     isGlobal(node.getExpression());
 }
 
 /**
- * @param {Node} node - the current ts-morph AST Node
+ * @param {Node} node - the ts-morph AST Node to evaluate
  *
  * @return {boolean} whether the node is a global set
  */
-function isGlobalSet(node) {
+function isSet(node) {
   return nodeTypes.isBinaryExpression(node) &&
-    node.getOperatorToken().getText() === '=' &&
-    isGlobalGet(node.getLeft());
+    node.getOperatorToken().getText() === '=';
+}
+
+/**
+ * @param {Node} node - the ts-morph AST Node to evaluate
+ *
+ * @return {boolean} whether the node is a window global
+ */
+function isWindowGlobal(node) {
+  // filter constructors
+  const baseConstructSignature = node.getType().getConstructSignatures()[0];
+  if (baseConstructSignature) {
+    return false;
+  }
+
+  // filter namespaces
+  if (node.getType && node.getType().getText() === node.getText()) {
+    return false;
+  }
+
+  const parent = node.getParent();
+  // filter data properties
+  const propertyName = node.getText();
+  if (parent.getExpression && parent.getExpression().getType() &&
+      isDataProperty(parent.getExpression().getType().getSymbol() &&
+      parent.getExpression().getType().getSymbol().getName(),
+      propertyName)) {
```

Comment:
This if repeats `parent.getExpression().getType()` several times.

In what cases is `parent.getExpression().getType()` falsy? In what cases is `parent.getExpression().getType().getSymbol()` falsy?

Avoid checking for `parent.getExpression`. Instead check the parent node type. I believe you want to check if it's a property access node?

I think the ideal code here would be

```js
const baseName = parent.getExpression().getType().getSymbol().getName();
```

inside an if that tests if it's a property access expression. But it's possible getType or getSymbol could fail. You need to comment on what cases that happens, though.

Reply:
In general I'm having a hard time understanding which case this is meant to cover, exactly. I suggest for all the filters, you add an example. E.g.

```js
// Filter constructors. Example: TypeError
// Filter namespaces. Example: CSS
// Filter data properties. Example: ...??
```

Reply:
Unless there are any `window` data properties I can simplify this by checking if the parent is a property access expression and just exiting out if so (no need to check `isDataProperty`)

Reply:
I'll go through and add some explanations on my filters and conditionals - it's likely some of them can be simplified